### PR TITLE
ENT-3750: Add Billing Type to Host Based Inventory

### PR DESCRIPTION
**Pull request by dmckoske**
Looking for feedback on the scope of changes for this card.  Testing instructions are forthcoming.

### CODE REVIEW

File: rhsm-client/src/main/java/org/candlepin/subscriptions/conduit/rhsm/client/StubRhsmApi.java [hunk 1]
```diff
@@ -58,6 +58,7 @@ public OrgInventory getConsumersForOrg(String xRhsmApiAccountID, Integer limit,
         consumer1.getFacts().put("uname.machine", "x86_64");
         consumer1.getFacts().put("virt.is_guest", "True");
         consumer1.getFacts().put("ocm.units", "Sockets");
+        consumer1.getFacts().put("billing_model", "standard");
```
Comment (kahowell) [inline rhsm-client/src/main/java/org/candlepin/subscriptions/conduit/rhsm/client/StubRhsmApi.java:61]:
```suggestion
        consumer1.getFacts().put("ocm.billing_model", "standard");
```

**Review by kahowell — CHANGES_REQUESTED** [review]
I would also expect to see changes to InventoryController to populate billingModel based on the value of "ocm.billing_model". You should follow the same pattern as what's InventoryController for "ocm.units" (call `setBillingModel` with the value from `rhsmFacts.get(OCM_BILLING_MODEL)`). InventoryControllerTest should also be updated accordingly.

In terms of testing, I'd suggest running with `RHSM_USE_STUB=true ./gradlew bootRun`, triggering a org sync via http://localhost:8080/actuator/hawtio/jmx/operations?nid=root-org.candlepin.subscriptions.conduit.jmx-RhsmConduitJmxBean-rhsmConduitJmxBean -> `syncOrg` (any orgId will do when using the stub), followed by viewing the `platform.inventory.host-ingress` topic via localhost:3030 (you should see the updated JSON containing `BILLING_MODEL`). This is what I'll do to review once my feedback is addressed :smile:

File: rhsm-client/src/main/java/org/candlepin/subscriptions/conduit/rhsm/client/StubRhsmApi.java (hunk 1 above)
Comment (dmckoske) [reply rhsm-client/src/main/java/org/candlepin/subscriptions/conduit/rhsm/client/StubRhsmApi.java]: I made the recommended changes, thanks for your help with this.  I will update the testing procedure tomorrow after I run it.

File: src/main/java/org/candlepin/subscriptions/conduit/inventory/InventoryService.java [hunk 2]
```diff
@@ -164,6 +164,7 @@ public OrgInventory getInventoryForOrgConsumers(List<ConduitFacts> conduitFactsF
         addFact(rhsmFactMap, "SYSPURPOSE_USAGE", conduitFacts.getSysPurposeUsage());
         addFact(rhsmFactMap, "SYSPURPOSE_ADDONS", conduitFacts.getSysPurposeAddons());
         addFact(rhsmFactMap, "SYSPURPOSE_UNITS", conduitFacts.getSysPurposeUnits());
+        addFact(rhsmFactMap, "OCM_BILLING_MODEL", conduitFacts.getBillingModel());
```
Comment (kahowell) [inline src/main/java/org/candlepin/subscriptions/conduit/inventory/InventoryService.java:167]: This should be just "BILLING_MODEL" for consistency w/ #406.

File: src/test/java/org/candlepin/subscriptions/conduit/inventory/DefaultInventoryServiceTest.java [hunk 3]
```diff
@@ -104,6 +105,7 @@ void testSendHostUpdatePopulatesAllFieldsWithFullConduitFactsRecord() throws Api
         expectedFactMap.put("SYSPURPOSE_USAGE", "test_usage");
         expectedFactMap.put("SYSPURPOSE_ADDONS", Arrays.asList("addon1", "addon2"));
         expectedFactMap.put("SYSPURPOSE_UNITS", "Sockets");
+        expectedFactMap.put("OCM_BILLING_MODEL", "standard");
```
Comment (kahowell) [inline src/test/java/org/candlepin/subscriptions/conduit/inventory/DefaultInventoryServiceTest.java:108]: Again, this should be just "BILLING_MODEL" for consistency w/ #406.

**Review by kahowell — CHANGES_REQUESTED** [review]
There are 2 places where the string "OCM_BILLING_MODEL" needs to be changed to "BILLING_MODEL", but this looks great otherwise!

**Review by lindseyburnett — APPROVED** [review]

**Review by kahowell — APPROVED** [review]

**Pull request merged**